src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSDoneFileMarker.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright © 2020-2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.gcs.actions;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.batch.BatchActionContext;
+import io.cdap.cdap.etl.api.batch.PostAction;
+import io.cdap.plugin.common.batch.action.Condition;
+import io.cdap.plugin.common.batch.action.ConditionConfig;
+import io.cdap.plugin.gcp.common.GCPConfig;
+import io.cdap.plugin.gcp.gcs.GCSPath;
+import io.cdap.plugin.gcp.gcs.StorageClient;
+import java.io.IOException;
+import javax.annotation.Nullable;
+
+/**
+ * A post action plugin that creates a marker file with a given name in case of a succeeded, failed or completed
+ * pipeline.
+ */
+@Plugin(type = PostAction.PLUGIN_TYPE)
+@Name(GCSDoneFileMarker.NAME)
+@Description("Creates a marker file with a given name in case of a succeeded, failed or completed pipeline.")
+public class GCSDoneFileMarker extends PostAction {
+  public static final String NAME = "GCSDoneFileMarker";
+  public Config config;
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    config.validate(pipelineConfigurer.getStageConfigurer().getFailureCollector());
+  }
+
+  @Override
+  public void run(BatchActionContext batchActionContext) throws IOException {
+    config.validate(batchActionContext.getFailureCollector());
+
+    if (!config.shouldRun(batchActionContext)) {
+      return;
+    }
+
+    Boolean isServiceAccountFilePath = config.isServiceAccountFilePath();
+
+    if (isServiceAccountFilePath == null) {
+      batchActionContext.getFailureCollector().addFailure("Service account type is undefined.",
```

Comment:
such validations should be done in `config.validate`

Reply:
Fixed. Please check @CuriousVini